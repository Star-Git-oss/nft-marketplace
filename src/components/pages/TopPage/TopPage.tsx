import React, { useEffect, useState } from 'react';
import { useTranslation } from 'react-i18next';
import axios from 'axios';
import { Helmet } from 'react-helmet';
import { ApiProxyUri } from '../../../utils/utils';
import Carousel from './Carousel/Carousel';
import { H1, H4, GradientText } from '../../common/Title/Title';

const TopPage: React.FC<LoadablePageType> = ({ setIsLoading }) => {

  const { t } = useTranslation();

  // Nft list in state
  const [nftList1, setNftList1] = useState([]);
  const [nftList2, setNftList2] = useState([]);
  const [nftList3, setNftList3] = useState([]);

  // // Retrieve NFT info when component loaded
  useEffect(() => {
    async function fetchData() {

      setIsLoading(true);

      const [res1, res2, res3] = await Promise.all(
        [
          axios.get(`${ApiProxyUri}/nfts`), 
          axios.get(`${ApiProxyUri}/nfts`),
          axios.get(`${ApiProxyUri}/nfts`),
        ]);
      
      setNftList1(res1.data.nfts);
      setNftList2(res2.data.nfts);
      setNftList3(res3.data.nfts);
      
      setIsLoading(false);
    }
    fetchData();
  }, [setIsLoading]);

  return (
    <>

      <Helmet>
        <title>{t('topPage.seo.title')}</title>
        <meta name="description" content={t('topPage.seo.description')} />
        <meta name="keywords" content={t('topPage.seo.keywords')} />
      </Helmet>
  
      <H1>
        {t('topPage.introduction1')}
        <GradientText>
          {t('topPage.introduction2')}
        </GradientText>
        {t('topPage.introduction3')}
      </H1>

      <H4>{t('topPage.categoryTitle')}</H4>
      {nftList1 && <Carousel big nftList={nftList1} />}

      <H4>{t('topPage.topCollector')}</H4>
      { nftList2 && <Carousel nftList={nftList2} />}
      
      <H4>{t('topPage.popularCreations')}</H4>
      { nftList3 && <Carousel nftList={nftList3} />}
    </>
  );
};

export default TopPage;
