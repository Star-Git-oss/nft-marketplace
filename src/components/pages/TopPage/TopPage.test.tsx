/* eslint-disable no-console */
import React from 'react';
import axios from 'axios';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import TopPage from './TopPage';

// Known issue: error and warning does not work for console
// https://github.com/facebook/react/issues/7047
const original = console.error;

beforeEach(() => {
  console.error = jest.fn();
});

afterEach(() => {
  console.error = original;
});

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key) => key }),
}));

const nftList = {
  nfts: [
    {
      'id':1,
      'labels': [1, 2, 3],
      'name': 'my NFT 1',
      'quantity': '10 of 10',
      'price': '0.2',
      'image': 'https://picsum.photos/200/300',
      'views': 5201,
      'owner': 'mickael canu',
      'creator': 'inri',
      'collectionName': 'Test name'
    },
    {
      'id':2,
      'labels': [1, 2, 3],
      'name': 'my NFT 2',
      'quantity': '10 of 10',
      'price': '0.5',
      'image': 'https://picsum.photos/201/300',
      'views': 5201,
      'owner': 'mickael canu',
      'creator': 'inri',
      'collectionName': 'Test name'
    },
  ]
};

// Axios mock
jest.mock('axios');
const resp = { data: nftList };
axios.get.mockResolvedValue(resp);

describe('TopPage snapshots', () => {


  it('TopPage renders correctly', () => {
    
    const tree = renderer
      .create(
        <TopPage setIsLoading={()=> jest.fn()} />
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });

});


describe('TopPage', () => {
  it('renders component properly', () => {

    const setIsLoading = jest.fn();
    render(<TopPage setIsLoading={setIsLoading} />);

    expect(screen.getByText('topPage.categoryTitle')).toBeInTheDocument();
    expect(screen.getByText('topPage.topCollector')).toBeInTheDocument();
    expect(screen.getByText('topPage.popularCreations')).toBeInTheDocument();

    // TODO fix this text.. nft info not reachable
    // await expect(screen.findByText('my NFT 1')).toBeVisible()();
    

  });

});