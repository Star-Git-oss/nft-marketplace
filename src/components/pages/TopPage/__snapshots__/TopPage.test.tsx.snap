// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopPage snapshots TopPage renders correctly 1`] = `
Array [
  <h1
    className="sc-jSgupP cLxsPC"
  >
    topPage.introduction1
    <span
      className="sc-iqHYGH gOpgzN"
    >
      topPage.introduction2
    </span>
    topPage.introduction3
  </h1>,
  <h4
    className="sc-fubCfw nLQFr"
  >
    topPage.categoryTitle
  </h4>,
  <div
    style={
      Object {
        "marginBottom": "40px",
      }
    }
  >
    <div
      className="react-multi-carousel-list  "
    >
      <ul
        className="react-multi-carousel-track "
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseMove={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "overflow": "unset",
            "transform": "translate3d(0px,0,0)",
            "transition": "none",
          }
        }
      />
    </div>
  </div>,
  <h4
    className="sc-fubCfw nLQFr"
  >
    topPage.topCollector
  </h4>,
  <div
    style={
      Object {
        "marginBottom": "40px",
      }
    }
  >
    <div
      className="react-multi-carousel-list  "
    >
      <ul
        className="react-multi-carousel-track "
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseMove={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "overflow": "unset",
            "transform": "translate3d(0px,0,0)",
            "transition": "none",
          }
        }
      />
    </div>
  </div>,
  <h4
    className="sc-fubCfw nLQFr"
  >
    topPage.popularCreations
  </h4>,
  <div
    style={
      Object {
        "marginBottom": "40px",
      }
    }
  >
    <div
      className="react-multi-carousel-list  "
    >
      <ul
        className="react-multi-carousel-track "
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseMove={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "overflow": "unset",
            "transform": "translate3d(0px,0,0)",
            "transition": "none",
          }
        }
      />
    </div>
  </div>,
]
`;
